src/index.ts

```diff
@@ -281,15 +283,26 @@ export default class DynamoDbDao<DataModel, KeySchema> {
    * Creates/Updates an item in the table
    */
   async put(data: DataModel, options: PutOptions = {}): Promise<DataModel> {
-    await this.documentClient
-      .put({
-        TableName: this.tableName,
-        Item: data,
-        ConditionExpression: options.conditionExpression,
-        ExpressionAttributeNames: options.attributeNames,
-        ExpressionAttributeValues: options.attributeValues,
-      })
-      .promise();
+    const payload = {
+      TableName: this.tableName,
+      Item: data,
+      ConditionExpression: options.conditionExpression,
+      ExpressionAttributeNames: options.attributeNames,
+      ExpressionAttributeValues: options.attributeValues,
+    };
+
+    // if the caller supplied returnValues, return them instead:
+    if (options.returnOldValues) {
+      const { Attributes: attributes } = await this.documentClient
+        .put({
+          ...payload,
+          ReturnValues: 'ALL_OLD',
+        })
+        .promise();
+      return (attributes ?? {}) as DataModel;
```

Comment:
Hmm, I'm not sure if I'm a fan of the casting being done here. This kinda violates the expectation that the full `DataModel` object will be returned from the call. 

I think I'd prefer to see the return type accurately reflect what can actually be returned.